Source/DedicatedServer/Private/ServerConsoleLinux.cpp

```diff
@@ -1,23 +1,244 @@
 // Copyright 2004-2016 YaS-Online, Inc. All Rights Reserved.
 
 #include "DedicatedServerPrivatePCH.h"
+#if PLATFORM_LINUX
+#include <iostream>
+#include <sstream>
+#include <string>
+#endif
 
 void DumpConsoleHelp();
 
 #if WITH_SERVER_CODE && PLATFORM_LINUX
 	void FServerConsole::Tick()
 	{
+		int hInputEvents = getch();
+		if ( hInputEvents != ERR )
+		{
+			if ( hInputEvents == '\n' )
+			{
+				auto& sInput = m_sInput;
+				AsyncTask( ENamedThreads::GameThread, [this, sInput]()
+				{
+					TCHAR sOutput[MAX_SPRINTF] = TEXT("");
+					if ( sInput.StartsWith( TEXT( "help" ) ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						DumpConsoleHelp();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add(sInput);
+
+						RedrawInputLine();
+					}
+					else if ( GEngine->Exec( GWorld, *sInput ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add( sInput );
+
+						RedrawInputLine();
+					}
+					else if ( GWorld->GetAuthGameMode() != nullptr && GWorld->GetAuthGameMode()->ProcessConsoleExec( *sInput, *GLog, nullptr ) )
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "> %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						m_hCommandHistory.Add( sInput );
+
+						RedrawInputLine();
+					}
+					else
+					{
+						ClearInputLine();
+						m_sInput.Empty();
+
+						attron( COLOR_PAIR( 1 ) );
+						FCString::Sprintf( sOutput, TEXT( "Unknown Command: %s%s" ), *sInput, LINE_TERMINATOR );
+						printf( "%s", TCHAR_TO_ANSI( sOutput ) );
+						refresh();
+						attroff( COLOR_PAIR( 1 ) );
+
+						RedrawInputLine();
+					}
+				});
+			}
+			else if ( hInputEvents == '\b' )
```

Comment:
Depending on the Linux machine you can have different input codes, in fact if I put KEY_BACKSPACE, the event is not triggered. Instead if I put the generic '\b' character, or '\n' character for KEY_ENTER, it will work on any Linux machine.